flink-core/src/main/java/org/apache/flink/api/common/io/ReplicatingInputFormat.java

```diff
@@ -59,7 +60,7 @@
  * @see org.apache.flink.api.common.operators.base.FilterOperatorBase
  * @see org.apache.flink.api.common.operators.base.MapPartitionOperatorBase
  */
-public final class ReplicatingInputFormat<OT, S extends InputSplit> implements InputFormat<OT, S> {
+public final class ReplicatingInputFormat<OT, S extends InputSplit> extends RichInputFormat<OT, S> {
```

Comment:
The `ReplicatingInputFormat` can wrap any kind of `InputFormat` (with or without RuntimeContext). 
Therefore, it needs to extend `RichInputFormat` but overwrite the `setRuntimeContext()` method and forward the context to its wrapped IF, if this is also a `RichInputFormat`.

Please check the updated JavaDocs as well. The wrapped IF does not need to be rich.